### DISCUSSION TITLE
How to integrate asset from Service Now CMBD

### DISCUSSION THREAD

**Discussion by rthreis404** [snipe/snipe-it#14113] [Q&A]
Hello everyone,
I'm all new to Snipe-It, and as far as I could see, it's a great product.
I'm asking my self a question, about the import / interaction available between ServiceNow cmdb and Snipe-It.
Explanation of what I plan to do.
We use the ServiceNow ticketing system, and a big part of the assets is also present in their cmdb.  Unfortunately, this is not precise enough for us.
In fact, we'd like to be able to manage monitors, keyboards, mice, docking stations, ... for the users.
We also would like to be able to send the user an email, in which he'll need to approve the assets he received on a delivery, the assets he returned on a leaving,...

But, instead of manually entering all the assets in Snipe-It, could you tell me if there's a way, using API perhaps, to "inject" the information from ServiceNow into Snipe-It ?  This way, we won't need an import / export procedure.  The assets would be updated automatically using the API call.
Or even, keep the the assets in Service Now, and update their information using the API over Snipe-It.
I'm not sure if my request is clear, I hope someone is going to be able to guide me in this.

Greetings,

**Comment by mauro-miatello** [comment]
I think you have only 2 options:
- manually export / import using csv
- create a task that update asset using API https://snipe-it.readme.io/reference/hardware-update

**Reply by rthreis404** [reply]
OK, no idea right now how I'm going to do it.
export / import is going to be a pain, because we got, in the production env, several thousands of assets...  In the test environment I'm setting up, just a couple of hundreds, but still.
So, I guess I'm going to put on my Developpercap again, and do some crazy API stuff 🤪
